docs/en/advanced_guides/datasets.md

```diff
@@ -0,0 +1,147 @@
+# Dataset
+
+In this document, we will introduce the design of each datasets in MMGeneration and how users to add their own dataset.
+
+In 1.x version of MMGeneration, all datasets are inherited from BaseDataset.
+Each dataset load the data list by `load_data_list` and call `prepare_data` in `__getitem__` to load the data info. In `prepare_data`, `get_data_info` and transforms pipeline will be called progressively.
+
+If you want to implement a new dataset class, you must implement `load_data_list` function. We also encourage users to use the original data loading logic provided by `BaseDataset`.
```

Comment:
you must implement `load_data_list` function
-->
you only need to implement `load_data_list` function